src/Symfony/Bundle/FrameworkBundle/FrameworkBundle.php

```diff
@@ -111,14 +112,24 @@ public function build(ContainerBuilder $container)
             KernelEvents::FINISH_REQUEST,
         ];
 
+        $noPreloadEvents = !class_exists(ConsoleEvents::class) ? [] : [
+            ConsoleEvents::COMMAND,
+            ConsoleEvents::TERMINATE,
+            ConsoleEvents::ERROR,
+        ];
+
         $container->addCompilerPass(new LoggerPass(), PassConfig::TYPE_BEFORE_OPTIMIZATION, -32);
         $container->addCompilerPass(new RegisterControllerArgumentLocatorsPass());
         $container->addCompilerPass(new RemoveEmptyControllerArgumentLocatorsPass(), PassConfig::TYPE_BEFORE_REMOVING);
         $container->addCompilerPass(new RoutingResolverPass());
         $container->addCompilerPass(new ProfilerPass());
         // must be registered before removing private services as some might be listeners/subscribers
         // but as late as possible to get resolved parameters
-        $container->addCompilerPass((new RegisterListenersPass())->setHotPathEvents($hotPathEvents), PassConfig::TYPE_BEFORE_REMOVING);
+        $container->addCompilerPass((new RegisterListenersPass())
+            ->setHotPathEvents($hotPathEvents)
+            ->setNoPreloadEvents($noPreloadEvents),
+            PassConfig::TYPE_BEFORE_REMOVING
```

Comment:
the indention looks pretty strange here. the method calls and the param shouldn't be on the same level.

Reply:
updated